src/PKSim.Core/Mappers/ExpressionProfileToExpressionProfileBuildingBlockMapper.cs

```diff
@@ -0,0 +1,90 @@
+﻿using OSPSuite.Core.Domain;
+using OSPSuite.Core.Domain.Builder;
+using OSPSuite.Core.Domain.Formulas;
+using OSPSuite.Core.Domain.Services;
+using OSPSuite.Utility;
+using PKSim.Core.Model;
+using IFormulaFactory = PKSim.Core.Model.IFormulaFactory;
+
+namespace PKSim.Core.Mappers
+{
+   public interface IExpressionProfileToExpressionProfileBuildingBlockMapper : IMapper<ExpressionProfile, ExpressionProfileBuildingBlock>
+   {
+   }
+
+   public class ExpressionProfileToExpressionProfileBuildingBlockMapper : IExpressionProfileToExpressionProfileBuildingBlockMapper
+   {
+      private readonly IObjectBaseFactory _objectBaseFactory;
+      private readonly IEntityPathResolver _entityPathResolver;
+      private IFormulaFactory _formulaFactory;
+
+      public ExpressionProfileToExpressionProfileBuildingBlockMapper(IObjectBaseFactory objectBaseFactory, IEntityPathResolver entityPathResolver, IFormulaFactory formulaFactory)
+      {
+         _objectBaseFactory = objectBaseFactory;
+         _entityPathResolver = entityPathResolver;
+         _formulaFactory = formulaFactory;
+      }
+
+      public ExpressionProfileBuildingBlock MapFrom(ExpressionProfile expressionProfile)
+      {
+         var expressionProfileBuildingBlock = _objectBaseFactory.Create<ExpressionProfileBuildingBlock>();
+
+         expressionProfileBuildingBlock.Name = expressionProfile.Name;
+         expressionProfileBuildingBlock.PKSimVersion = ProjectVersions.Current.VersionDisplay;
```

Comment:
Display is also not good. WE do not have a one to one mapping. We only increment the version if we have a serialization change
You need to grab the ApplicationConfiugration or sethg like this and get the version from there.

Reply:
probably this one here
https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/blob/develop/src/OSPSuite.Core/IApplicationConfiguration.cs#L45

you can see what we export in the PopulationExporTTask when we dump a population to csv. This should be the same logic